### ISSUE TITLE
Add Dockerfile for containerization

### ISSUE THREAD

**Issue by CodexRaunak** [AkshitLakhera/PenCraft-Full-Stack-Blogging-Application#86]
#### We need to add a Dockerfile to containerize the PenCraft-Full-Stack-Blogging-Application. This will help ensure consistency across different development and production environments. The Dockerfile should:

1. Use a Node.js image as the base.
2. Set the working directory.
3. Install dependencies for the backend, frontend, and common packages.
4. Build the frontend using the vite build tool.
5. Copy the necessary files.
6. Expose port 3000 and run the backend development server (wrangler dev).

**Acceptance Criteria:**

1. Dockerfile is added to the root directory of the project.
2. Dockerfile can successfully build and run the application.

**Comment by CodexRaunak** [comment]
can I work on it @AkshitLakhera under gssoc-24ext and hacktoberfest, i am interested in solving this issue

**Comment by Ameerjafar** [comment]
@AkshitLakhera is the issue it still open.

**Comment by CodexRaunak** [comment]
Yes, I am working on it, i will finish it by today and make the PR, sorry for the delay, i had an exam

**Comment by CodexRaunak** [comment]
![image](https://github.com/user-attachments/assets/47b6df89-b5ff-4b9a-a0ee-05a414f4b1f2)
Can anyone help me 
@AkshitLakhera 
with this error logs upon running docker-compose up --build
I created 2 docker files, one for the frontend and one for the backend 

![image](https://github.com/user-attachments/assets/111a0c18-85eb-4221-b18b-e207b17e212e)
![image](https://github.com/user-attachments/assets/922b926a-28dd-41be-a2e5-d332becdb281)
